RedditSharp/Reddit.cs

```diff
@@ -229,7 +229,7 @@ public void InitOrUpdateUser()
             var response = (HttpWebResponse)request.GetResponse();
             var result = WebAgent.GetResponseString(response.GetResponseStream());
             var json = JObject.Parse(result);
-            User = new AuthenticatedUser().Init(this, json, WebAgent);
+            User = new AuthenticatedUser().Init(this, json, WebAgent).Result;
```

Comment:
Shouldn't be using .Result , means the method probably got missed when you were reverting things